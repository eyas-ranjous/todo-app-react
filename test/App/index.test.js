import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'
import createStore from '../../app/store';
import TodoApp from '../../app/App';

describe('TodoApp', () => {
  const store = createStore();

  it('renders successfully', () => {
    const wrapper = mount(
      <Provider store={store}>
        <TodoApp />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
