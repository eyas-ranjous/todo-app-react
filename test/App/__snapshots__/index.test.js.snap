// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp renders successfully 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <_default>
    <div
      className="todo-app"
    >
      <h1>
        React/Redux/Hooks Todo App
      </h1>
      <TodoInput>
        <div
          className="todo-input"
        >
          <TextInput>
            <input
              className="todo-text-input"
              onKeyDown={[Function]}
              placeholder="Write Todo Text..."
              type="text"
            />
          </TextInput>
          <AddButton>
            <button
              onClick={[Function]}
            >
              Add
            </button>
          </AddButton>
        </div>
      </TodoInput>
      <Component>
        <p
          className="no-todos-msg"
        >
          No Todos Found
        </p>
      </Component>
      <TodoFilters>
        <div>
          <AllFilterButton
            setFilter={[Function]}
          >
            <button
              className="filter-button filter-button-all filter-button-selected"
              onClick={[Function]}
            >
              All
            </button>
          </AllFilterButton>
          <TodoFilterButton
            setFilter={[Function]}
          >
            <button
              className="filter-button filter-button-todo"
              onClick={[Function]}
            >
              To Do
            </button>
          </TodoFilterButton>
          <DoneFilterButton
            setFilter={[Function]}
          >
            <button
              className="filter-button filter-button-done"
              onClick={[Function]}
            >
              Done
            </button>
          </DoneFilterButton>
        </div>
      </TodoFilters>
    </div>
  </_default>
</Provider>
`;
